src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -796,6 +817,57 @@ void Transformations::MainSnippets(void) {
     };
 #endif // OPENVINO_ARCH_X86_64
 
+    auto is_support_op = [](const std::shared_ptr<const ov::Node> &n) -> bool {
+#if defined(OPENVINO_ARCH_ARM64)
+        return (ov::is_type<ov::op::v1::Add>(n) ||
+                ov::is_type<ov::op::v1::Multiply>(n) ||
+                ov::is_type<ov::op::v0::Relu>(n));
+#else
+        // CPU Plugin support Swish in Subgraph via conversion to SwichCPU which assumes second input to be constant
+        auto is_unsupported_swish = [](const std::shared_ptr<const ov::Node> &n) {
+            return ov::is_type<const ov::op::v4::Swish>(n) && n->inputs().size() > 1 &&
+                   !ov::is_type<const ov::op::v0::Constant>(n->get_input_node_shared_ptr(1));
+        };
+        // todo: general tokenization flow is not currently supported for these operations.
+        // they can be tokenized only as a part of complex patterns
+        auto is_unsupported_by_common_tokenization = [](const std::shared_ptr<const ov::Node> &n) {
+            return (ov::is_type<const ov::op::v1::Softmax>(n) ||
+                    ov::is_type<const ov::op::v8::Softmax>(n) ||
+                    ov::is_type<const ov::op::v0::MatMul>(n) ||
+                    ov::is_type<const ov::op::v1::Transpose>(n) ||
+                    ov::is_type<const ov::op::v1::Broadcast>(n) ||
+                    ov::is_type<const ov::op::v3::Broadcast>(n) ||
+                    ov::is_type<const ov::op::v1::ReduceMax>(n) ||
+                    ov::is_type<const ov::op::v1::ReduceSum>(n));
+        };
+        return !is_unsupported_swish(n) && !is_unsupported_by_common_tokenization (n);
+#endif
+    };
+
+    auto has_supported_element_types = [](const std::shared_ptr<const ov::Node> &n) -> bool {
+        auto supported = [&n](descriptor::Tensor& t) -> bool {
```

Comment:
Applied. Thanks Alexandra!

Reply:
I meant to create lamda function `is_supported_tensor` that will contains all tensors checks: precision, ranks 😄 Something like:
```
auto has_supported_tensors = [](const std::shared_ptr<const ov::Node> &n) -> bool {
        // Check for supported precision
        auto is_supported_tensor = [&n](descriptor::Tensor& t, bool is_input) -> bool {
                         // TODO [105804] int32 isn't supported in general because i32 emitters are required for bit-exact i32 calculations in some cases
                         //  So i32 is supported exclusively for transposes and broadcast
                static const std::set<ov::element::Type> supported_element_types =
#if defined(OPENVINO_ARCH_ARM64)
                        { ov::element::f32 };
#else
                         { ov::element::f32, ov::element::bf16, ov::element::i8, ov::element::u8 };
#endif

                 bool is_supported_element_type = (supported_element_types.count(t.get_element_type()) != 0 || is_input &&  (t.get_element_type() == ov::element::i32 && (ov::is_type<const opset1::Transpose>(n) ||  ov::is_type<const opset1::Broadcast>(n)));
                 
                 // Check for supported ranks
                 // todo: clarify whether we can evaluate snippets on inputs with larger ranks
                  bool is_supported_rank =  t.get_partial_shape().rank().get_length() <= 6;
                 return is_supported_element_type && is_supported_rank ;
        };

        const auto& inputs = n->inputs();
        const auto& outputs = n->outputs();
        return std::all_of(inputs.begin(), inputs.end(), [&](const Input<const ov::Node>& in) {return  supported(in.get_tensor());}) &&
                   std::all_of(outputs.begin(), outputs.end(), [&](const Output<const ov::Node>& out) {return  supported(out.get_tensor());});
};
```

Reply:
Applied. Thanks Alexandra!